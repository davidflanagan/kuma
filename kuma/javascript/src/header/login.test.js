//@flow
import React from 'react';
import { act, create } from 'react-test-renderer';
import DocumentProvider from '../document-provider.jsx';
import { fakeDocumentData } from '../document-provider.test.js';
import Dropdown from './dropdown.jsx';
import Login from './login.jsx';
import UserProvider from '../user-provider.jsx';

test('Login snapshot before user data is fetched', () => {
    const login = create(
        <DocumentProvider initialDocumentData={fakeDocumentData}>
            <UserProvider.context.Provider value={null}>
                <Login />
            </UserProvider.context.Provider>
        </DocumentProvider>
    ).toJSON();
    expect(login).toBe(null);
});

test('Login component when user is not logged in', () => {
    const login = create(
        <DocumentProvider initialDocumentData={fakeDocumentData}>
            <UserProvider.context.Provider value={UserProvider.defaultUserData}>
                <Login />
            </UserProvider.context.Provider>
        </DocumentProvider>
    ).toJSON();
    expect(login.children[0]).toBe('Sign in');
    expect(login).toMatchSnapshot();
});

test('Login component when user is logged in', () => {
    const login = create(
        <DocumentProvider initialDocumentData={fakeDocumentData}>
            <UserProvider.context.Provider
                value={{
                    ...UserProvider.defaultUserData,
                    isAuthenticated: true,
                    username: 'test-username',
                    gravatarUrl: { small: 'test-url', large: 'test-bigurl' }
                }}
            >
                <Login />
            </UserProvider.context.Provider>
        </DocumentProvider>
    );

    expect(login.toJSON()).toMatchSnapshot();

    // Expect a Dropdown element
    let root = login.root;
    let dropdown = root.findByType(Dropdown);
    expect(dropdown).toBeDefined();

    // Whose label prop is an image with the expected src and alt attributes
    expect(dropdown.props.label.props.srcSet).toContain('test-url');
    expect(dropdown.props.label.props.srcSet).toContain('test-bigurl');
    expect(dropdown.props.label.props.alt).toEqual('test-username');

    // Open up the dropdown menu
    act(() => {
        login.toJSON().children[0].children[0].props.onClick();
    });

    expect(login.toJSON()).toMatchSnapshot();

    let string = JSON.stringify(login.toJSON());
    expect(string).toContain('View profile');
    expect(string).toContain('Edit profile');
    expect(string).toContain('Sign out');
});
