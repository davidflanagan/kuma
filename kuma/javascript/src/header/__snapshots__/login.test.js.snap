// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component when user is logged in 1`] = `
.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 24px;
}

.emotion-8 button {
  font-size: 1em;
}

.emotion-6 {
  position: relative;
  pointer-events: auto;
  margin-left: -5px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  padding: 0 5px;
  border: none;
}

.emotion-1:hover {
  background-color: #eee;
}

.emotion-1:focus {
  outline-offset: -3px;
}

.emotion-0 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
}

.emotion-4 {
  position: absolute;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  background-color: white;
  border: solid #83d0f2 1.5px;
  box-shadow: 3px 3px 5px rgba(0,0,0,0.25);
  padding: 10px;
  min-width: 100%;
  right: 0;
  display: none;
}

.emotion-4 li {
  padding: 5px;
  white-space: nowrap;
}

.emotion-3 {
  border-width: 0;
  padding: 0;
  color: #3d7e9a;
  font-size: 1em;
  font-weight: normal;
}

.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-6 emotion-7"
  >
    <button
      className="emotion-1 emotion-2"
      onClick={[Function]}
    >
      <img
        alt="test-username"
        className="emotion-0"
        src="/static/img/avatar.png"
        srcSet="test-bigurl 200w test-url 50w"
      />
    </button>
    <ul
      className="dropdown-menu emotion-4 emotion-5"
    >
      <li>
        <a
          href="/en-US/profiles/test-username"
        >
          View profile
        </a>
      </li>
      <li>
        <a
          href="/en-US/profiles/test-username/edit"
        >
          Edit profile
        </a>
      </li>
      <li>
        <form
          action="/en-US/users/signout"
          method="post"
        >
          <input
            name="next"
            type="hidden"
            value="[fake absolute url]"
          />
          <button
            className="emotion-3"
            type="submit"
          >
            Sign out
          </button>
        </form>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Login component when user is logged in 2`] = `
.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 24px;
}

.emotion-8 button {
  font-size: 1em;
}

.emotion-6 {
  position: relative;
  pointer-events: auto;
  margin-left: -5px;
}

.emotion-0 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
}

.emotion-3 {
  border-width: 0;
  padding: 0;
  color: #3d7e9a;
  font-size: 1em;
  font-weight: normal;
}

.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  padding: 0 5px;
  border: none;
  background-color: #83d0f2;
}

.emotion-1:hover {
  background-color: #eee;
}

.emotion-1:focus {
  outline-offset: -3px;
}

.emotion-1:hover {
  background-color: #83d0f2;
}

.emotion-4 {
  position: absolute;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  background-color: white;
  border: solid #83d0f2 1.5px;
  box-shadow: 3px 3px 5px rgba(0,0,0,0.25);
  padding: 10px;
  min-width: 100%;
  right: 0;
}

.emotion-4 li {
  padding: 5px;
  white-space: nowrap;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-6 emotion-7"
  >
    <button
      className="emotion-1 emotion-2"
      onClick={null}
    >
      <img
        alt="test-username"
        className="emotion-0"
        src="/static/img/avatar.png"
        srcSet="test-bigurl 200w test-url 50w"
      />
    </button>
    <ul
      className="dropdown-menu emotion-4 emotion-5"
    >
      <li>
        <a
          href="/en-US/profiles/test-username"
        >
          View profile
        </a>
      </li>
      <li>
        <a
          href="/en-US/profiles/test-username/edit"
        >
          Edit profile
        </a>
      </li>
      <li>
        <form
          action="/en-US/users/signout"
          method="post"
        >
          <input
            name="next"
            type="hidden"
            value="[fake absolute url]"
          />
          <button
            className="emotion-3"
            type="submit"
          >
            Sign out
          </button>
        </form>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Login component when user is not logged in 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1em;
  font-weight: bold;
  color: black;
  margin-left: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #eee;
}

.emotion-0:focus {
  outline-offset: -3px;
}

<a
  className="emotion-0"
  data-service="GitHub"
  href="/users/github/login/?next=[fake absolute url]"
  rel="nofollow"
>
  Sign in
</a>
`;
